homeassistant/components/iron_os/coordinator.py

```diff
@@ -16,24 +18,43 @@
 _LOGGER = logging.getLogger(__name__)
 
 SCAN_INTERVAL = timedelta(seconds=5)
+SCAN_INTERVAL_GITHUB = timedelta(minutes=60)
 
 
-class IronOSCoordinator(DataUpdateCoordinator[LiveDataResponse]):
-    """IronOS coordinator."""
+class IronOSBaseCoordinator[_DataT](DataUpdateCoordinator[_DataT]):
+    """IronOS base coordinator."""
 
     device_info: DeviceInfoResponse
     config_entry: ConfigEntry
 
-    def __init__(self, hass: HomeAssistant, device: Pynecil) -> None:
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        device: Pynecil,
+        update_interval: timedelta,
+    ) -> None:
         """Initialize IronOS coordinator."""
         super().__init__(
             hass,
             _LOGGER,
             name=DOMAIN,
-            update_interval=SCAN_INTERVAL,
+            update_interval=update_interval,
         )
         self.device = device
 
+    async def _async_setup(self) -> None:
+        """Set up the coordinator."""
+
+        self.device_info = await self.device.get_device_info()
```

Comment:
would it make sense to do this in init.py and make it a parameter to avoid to call it twice?

Reply:
Isn't `_async_setup` only called once during initialization? 🤔

Oh, right, it is called once per coordinator initialization. Good point!

Reply:
Thinking about it, now that firmware version is relevant for the update entity, it maybe be necessary to refresh the device info from time to time, in case it has been updated in the meantime, not only on integration setup 🤔

Reply:
Then you should move it to the update function

Reply:
Ok, never mind, just remembered the device_info is cached in the library, so calling it twice would not result in fetching the characteristics from the device again.